quickstart_template/101-spinnaker-aks.sh

```diff
@@ -0,0 +1,161 @@
+  local value="$2"
+  if [ -z "$value" ]; then
+    echo "Parameter '$name' cannot be empty." 1>&2
+    print_usage
+    exit -1
+  fi
+}
+
+function run_util_script() {
+  local script_path="$1"
+  shift
+  curl --silent "${artifacts_location}${script_path}${artifacts_location_sas_token}" | sudo bash -s -- "$@"
+  local return_value=$?
+  if [ $return_value -ne 0 ]; then
+    >&2 echo "Failed while executing script '$script_path'."
+    exit $return_value
+  fi
+}
+
+function install_az() {
+  if !(command -v az >/dev/null); then
+    sudo apt-get update && sudo apt-get install -y libssl-dev libffi-dev python-dev
+    echo "deb [arch=amd64] https://apt-mo.trafficmanager.net/repos/azure-cli/ wheezy main" | sudo tee /etc/apt/sources.list.d/azure-cli.list
+    sudo apt-key adv --keyserver apt-mo.trafficmanager.net --recv-keys 417A0893
+    sudo apt-get install -y apt-transport-https
+    sudo apt-get -y update && sudo apt-get install -y azure-cli --allow-unauthenticated
+  fi
+}
+
+# Set defaults
+artifacts_location="https://raw.githubusercontent.com/azure/azure-devops-utils/master/"
+artifacts_location_sas_token=""
+
+while [[ $# > 0 ]]
+do
+  key="$1"
+  shift
+  case $key in
+    --app_id|-ai)
+      app_id="$1";;
+    --app_key|-ak)
+      app_key="$1";;
+    --username|-u)
+      username="$1";;
+    --password|-p)
+      password="$1";;
+    --tenant_id|-ti)
+      tenant_id="$1";;
+    --subscription_id|-si)
+      subscription_id="$1";;
+    --resource_group|-rg)
+      resource_group="$1";;
+    --vault_name|-vn)
+      vault_name="$1";;
+    --storage_account_name|-san)
+      storage_account_name="$1";;
+    --storage_account_key|-sak)
+      storage_account_key="$1";;
+    --aksClusterName|-aks)
+      aksClusterName="$1";;
+    --region|-r)
+      region="$1";;
+    --artifacts_location|-al)
+      artifacts_location="$1";;
+    --sas_token|-st)
+      artifacts_location_sas_token="$1";;
+    --help|-help|-h)
+      print_usage
+      exit 13;;
+    *)
+      echo "ERROR: Unknown argument '$key' to script '$0'" 1>&2
+      exit -1
+  esac
+  shift
+done
+
+throw_if_empty app_id $app_id
+throw_if_empty app_key $app_key
+throw_if_empty username $username
+throw_if_empty password $password
+throw_if_empty tenant_id $tenant_id
+throw_if_empty subscription_id $subscription_id
+throw_if_empty resource_group $resource_group
+throw_if_empty vault_name $vault_name
+throw_if_empty storage_account_name $storage_account_name
+throw_if_empty storage_account_key $storage_account_key
+throw_if_empty region $region
+throw_if_empty aksClusterName $aksClusterName
+
+#install az and hal
+install_az
+run_util_script "spinnaker/install_halyard/install_halyard.sh" -san "$storage_account_name" -sak "$storage_account_key" -u "$username"
+
+#get-credentials from aks
+az login --service-principal -u $app_id -p $app_key -t $tenant_id 
+az aks get-credentials --resource-group $resource_group --name $aksClusterName -f /home/$username/.kube/config 
```

Comment:
Per https://docs.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#az-aks-get-credentials, the default “config file” has already been  “~/.kube/config”, so assume the “-f xxx” is unnecessary?

Reply:
Script is executed with root privileges so kubeconfig will in /root/.kube by default,however halconfig will in /home/[username]/.hal becuse it must install with a user not root. So I need to change the path of .kube